### PR TITLE
Refactor/redesign group tiles

### PR DESCRIPTION
**Pull request by nicole-eb**

### CODE REVIEW

File: packages/renderers/lib/src/request/request_item_group_renderer.dart [hunk 1]
```diff
@@ -35,26 +35,41 @@ class RequestItemGroupRenderer extends StatelessWidget {
...
-      initiallyExpanded: true,
-      // TODO: render anything else than empty string when title is not defined
-      title: Text(requestItemGroup.title ?? ''),
-      subtitle: requestItemGroup.description != null ? Text(requestItemGroup.description!) : null,
-      children: requestItems,
+    return Container(
+      color: Theme.of(context).colorScheme.surfaceContainer,
+      child: ExpansionTile(
+          maintainState: true,
+          minTileHeight: 75,
```
Comment (Siolto) [inline packages/renderers/lib/src/request/request_item_group_renderer.dart:58]: @jkoenig134: is there a better way than hardcoding this? according to the design it should be 75

Comment (jkoenig134) [reply packages/renderers/lib/src/request/request_item_group_renderer.dart:58]: what's the issue with hardcoding?

Comment (Siolto) [reply packages/renderers/lib/src/request/request_item_group_renderer.dart:58]: not sure 😄  Hardcoding heights and widths always feels a little bit weird when it comes to responsive design^^

Comment (jkoenig134) [reply packages/renderers/lib/src/request/request_item_group_renderer.dart:58]: I think it's ok as it is only the minHeight

**Review by Siolto — DISMISSED** [review]

File: packages/renderers/lib/src/request/request_item_group_renderer.dart [hunk 2]
```diff
@@ -35,26 +35,41 @@ class RequestItemGroupRenderer extends StatelessWidget {
...
 
-    return ExpansionTile(
-      maintainState: true,
-      initiallyExpanded: true,
-      // TODO: render anything else than empty string when title is not defined
-      title: Text(requestItemGroup.title ?? ''),
-      subtitle: requestItemGroup.description != null ? Text(requestItemGroup.description!) : null,
-      children: requestItems,
+    return Container(
+      color: Theme.of(context).colorScheme.surfaceContainer,
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_group_renderer.dart:55]: why not use `backgroundColor` and `collapsedBackgroundColor ` from ExpansionTile?

File: packages/renderers/lib/src/request/request_item_group_renderer.dart [hunk 3]
```diff
@@ -35,26 +35,41 @@ class RequestItemGroupRenderer extends StatelessWidget {
...
-        itemIndex: (rootIndex: itemIndex.rootIndex, innerIndex: index),
-        controller: controller,
-        requestStatus: requestStatus,
-        openAttributeSwitcher: openAttributeSwitcher,
-        currentAddress: currentAddress,
-        expandFileReference: expandFileReference,
-        chooseFile: chooseFile,
-        openFileDetails: openFileDetails,
+      return Container(
+        color: Theme.of(context).colorScheme.surface,
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_group_renderer.dart:39]: why do we need a color here?

File: packages/renderers/lib/src/request/request_item_group_renderer.dart [hunk 4]
```diff
@@ -35,26 +35,41 @@ class RequestItemGroupRenderer extends StatelessWidget {
...
+          subtitle: requestItemGroup.description != null
+              ? Text(
+                  requestItemGroup.description!,
+                  style: Theme.of(context).textTheme.bodyMedium,
+                  maxLines: 2,
+                  overflow: TextOverflow.ellipsis,
+                )
+              : null,
+          iconColor: Theme.of(context).colorScheme.onSurfaceVariant,
+          children: requestItems),
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_group_renderer.dart:72]: missing trailing comma...

File: packages/renderers/lib/src/request/request_item_group_renderer.dart (hunk 2 above)
Comment (Siolto) [reply packages/renderers/lib/src/request/request_item_group_renderer.dart:55]: argh. haven't seen the `collapsedBackgroundColor` property...

File: packages/renderers/lib/src/request/request_item_group_renderer.dart (hunk 3 above)
Comment (Siolto) [reply packages/renderers/lib/src/request/request_item_group_renderer.dart:39]: currently there is only a Padding widget and then the individual renderers. I could move the container inside of the `RequestItemRenderer`

Comment (jkoenig134) [reply packages/renderers/lib/src/request/request_item_group_renderer.dart:39]: Hmm. Please make a ColoredBox out of it and leave it where it is for now.

Comment (jkoenig134) [reply packages/renderers/lib/src/request/request_item_group_renderer.dart:39]: oh - even better:

image

File: packages/renderers/lib/src/request/request_item_renderer/request_item_renderer.dart [hunk 5]
```diff
@@ -21,23 +21,26 @@ class RequestItemRenderer extends StatelessWidget {
...
+      required this.item,
+      required this.itemIndex,
+      this.controller,
+      this.requestStatus,
+      this.isRejected = false,
+      this.openAttributeSwitcher,
+      required this.expandFileReference,
+      required this.chooseFile,
+      required this.openFileDetails,
+      this.backgroundColor});
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_renderer/request_item_renderer.dart:38]: @Siolto please pay more attention to trailing commas, this must be resolved in your self review

File: packages/renderers/lib/src/request/request_item_group_renderer.dart [hunk 6]
```diff
@@ -45,15 +45,28 @@ class RequestItemGroupRenderer extends StatelessWidget {
...
       initiallyExpanded: true,
+      shape: const LinearBorder(side: BorderSide.none),
       // TODO: render anything else than empty string when title is not defined
       title: Text(requestItemGroup.title ?? ''),
-      subtitle: requestItemGroup.description != null ? Text(requestItemGroup.description!) : null,
+      subtitle: requestItemGroup.description != null
+          ? Text(
+              requestItemGroup.description!,
+              style: Theme.of(context).textTheme.bodyMedium,
+              maxLines: 2,
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_group_renderer.dart:65]: IMO maxLines should not be set here. The user has no possibility to access the text if it is truncated.

File: packages/renderers/lib/src/request/request_item_group_renderer.dart [hunk 7]
```diff
@@ -45,15 +45,21 @@ class RequestItemGroupRenderer extends StatelessWidget {
...
+      collapsedBackgroundColor: Theme.of(context).colorScheme.surfaceContainer,
       maintainState: true,
+      minTileHeight: 75,
       initiallyExpanded: true,
+      shape: const LinearBorder(side: BorderSide.none),
       // TODO: render anything else than empty string when title is not defined
       title: Text(requestItemGroup.title ?? ''),
-      subtitle: requestItemGroup.description != null ? Text(requestItemGroup.description!) : null,
+      subtitle: requestItemGroup.description != null ? Text(requestItemGroup.description!, style: Theme.of(context).textTheme.bodyMedium) : null,
+      iconColor: Theme.of(context).colorScheme.onSurfaceVariant,
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_group_renderer.dart:62]: shouldn't all the Text and Icon colors be `onSurfaceContainer`?

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**